File: src/ffi/alerts/piece_finished.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type piece_finished_alert = crate::ffi::ffi::piece_finished_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/piece_finished.h");
        type PieceIndex = crate::ffi::ffi::PieceIndex;
        unsafe fn piece_finished_alert_get_piece_index(
            alert: *mut piece_finished_alert,
        ) -> PieceIndex;
    }
}

File: cpp/alerts/piece_finished.cpp
#include "./piece_finished.h"
#include <cstdint>

namespace ltrs {

    PieceIndex piece_finished_alert_get_piece_index(lt::piece_finished_alert* alert) {
        return PieceIndex { .inner = static_cast<int32_t>(alert->piece_index) };
    }

} // namespace ltrs
